### DISCUSSION TITLE
Override Laravel Nova Vue Components (e.g. MainHeader)

### DISCUSSION THREAD

**Discussion by CBreetzi** [laravel/nova-issues#5795] [Ideas & Feature Requests] [answered]
Dear Laravel Nova Team,

thank you for the time and dedication to the project!
In Laravel Nova 4 I haven't found an elegant way to override the MainHeader and other similar core components.
I stumbled upon a solution (https://laracasts.com/discuss/channels/nova/how-to-overwrite-nova-vue-files) that tells me to register the component in a tool, but this doesn't work, since a local import / component registration can't be overrided by a global import. 
So I would have to change the Vue files in the vendor directory, which is not a good practice at all. Alternatively there's also this composer package (https://github.com/Muetze42/nova-assets-changer) that does the same thing in an easier way, but still kinda hacky.

Is there an ETA or plan for a feature that enables us to override Laravel's core components?

Thank you!

Cheers,
Chris

**Comment by davidhemphill** [comment, marked as answer]
Most Nova components can be overridden in the manner found in the forum post, but Nova's main header and footer cannot be overridden at this time.

**Reply by HugoHeneault** [reply]
It would be great having this information in doc in customization view ;)

**Reply by davidhemphill** [reply]
The reason it's not documented is because we don't support or recommend users replace Nova's internal components. :-)
